lib/Conversion/TorchToTcp/Utils.cpp

```diff
@@ -0,0 +1,57 @@
+//===------------------------------------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// Also available under a BSD-style license. See LICENSE.
+//
+//===----------------------------------------------------------------------===//
+
+#include "Utils.h"
+
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/Tensor/IR/Tensor.h"
+#include "torch-mlir-dialects/Dialect/Tcp/IR/TcpDialect.h"
+#include "torch-mlir-dialects/Dialect/Tcp/IR/TcpOps.h"
+
+using namespace mlir;
+using namespace mlir::tcp;
+
+Value torch_to_tcp::broadcastRankInLeadingDims(
+                                ConversionPatternRewriter &rewriter,
+                                Value input,
+                                int64_t rankIncrease) {
+  RankedTensorType inputType = input.getType().dyn_cast<RankedTensorType>();
+
+  SmallVector<ReassociationExprs, 4> reassociationMap(inputType.getRank());
```

Comment:
nit: remove `4`, since the actual size of this vector could be anything. See: https://llvm.org/docs/ProgrammersManual.html#llvm-adt-smallvector-h